File: backend/Models/Entities/Artist.cs
using Models.Base;

namespace Models.Entities
{
    public class Artist : BaseEntity
    {
        public string UserId { get; set; }
        public virtual User User { get; set; }
        public virtual List<Album> Albums { get; set; }
        public virtual List<Track> Tracks { get; set; }

        public Artist()
        {
            Albums = new List<Album>();
            Tracks = new List<Track>();
        }
    }
}

File: Frontend/admin_app/lib/models/artist.dart
import 'package:admin_app/models/base/base_entity.dart';
import 'package:admin_app/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'artist.g.dart';

@JsonSerializable()
class Artist extends BaseEntity {
  String? userId;
  User? user;

  Artist({
    required super.id,
    required super.createdAt,
    required super.updatedAt,
    required this.userId,
    required this.user,
  });

  factory Artist.fromJson(Map<String, dynamic> json) => _$ArtistFromJson(json);
  
  Map<String, dynamic> toJson() => _$ArtistToJson(this);
}
